File: array/twopointer/squaresSorted.cpp
#include <iostream>
#include <vector>
#include <math.h>
using namespace std;
vector<int> squaresSorted(vector<int> nums){
      int l=0,r=nums.size()-1,idx=0;
      vector<int> res;
      while(l<=r){
            if(abs(nums[l])>abs(nums[r])){
                  res.insert(res.begin(),pow(nums[l],2));
                  l++;
            }
            else{
                  res.insert(res.begin(),pow(nums[r],2));
                  r--;
            }
            
      }
      return res;
}
int main(){
      vector<int> n={-7,-3,2,3,11};
      vector<int> res=squaresSorted(n);
      for(int i: res) cout << i << " ";

}

File: array/twopointer/SquaresSorted.java
class SquaresSorted{
      public static int[] squaresSorted(int[] nums){
            int[] result = new int[nums.length];
            int l=0,r=nums.length-1;
            int idx=r;
      
            while(l<=r){
                  if(Math.abs(nums[r])>=Math.abs(nums[l]))  result[idx--] = nums[r]*nums[r--];

                  else result[idx--] = nums[l]*nums[l++];
            }
        

            return result;
      }
      public static void main(String[] args) {
          int[] n = {-4,-1,0,3,10};
          int[] res = squaresSorted(n);

          for(int x: res){
            System.out.println(x+" ");
          }
      }
}